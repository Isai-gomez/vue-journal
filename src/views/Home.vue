<template>
  <div class="home">
    <button @click="getToDayBook" class="btn btn-primary">{{doesToUpperCase}}</button>
  </div>
</template>

<script>
// @ is an alias to /src

export default {
  name: 'Home',
  components: {},
  data(){
    const titleBtn='Journal day'
    return{
      titleBtn
    }
  },
  methods: {
    getToDayBook() {
      this.$router.push({ name: 'no-entry' })
    },

  },
  computed:{
    doesToUpperCase:function(){
      return this.titleBtn.toUpperCase()
    }
  }
}
</script>
<style lang="scss" scoped>
.home{
  display: grid;
  height: 100vh;
  place-content: center;
}
</style>
