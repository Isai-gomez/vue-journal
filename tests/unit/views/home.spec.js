import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'

describe('Pruba en el Home vue', () => {
  test('Debe de rendizar el componente correctamente', () => {
    const wrapper = shallowMount(Home)
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('Al dar click en boton debe enviar a la ruata no-entry', () => {
    const moskRoute = {
      push: jest.fn(),
    }
    const wrapper = shallowMount(Home, {
      global: {
        mocks: {
          $router: moskRoute,
        },
      },
    })
    wrapper.find('button').trigger('click')
    expect(moskRoute.push).toHaveBeenCalled()
    expect(moskRoute.push).toHaveBeenCalledWith({ name: 'no-entry' })
  })
})
