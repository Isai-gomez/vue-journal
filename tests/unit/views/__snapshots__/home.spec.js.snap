// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruba en el Home vue Debe de rendizar el componente correctamente 1`] = `<div class="home"><button class="btn btn-primary">primary</button><button class="btn btn-secondary">secondary</button><button class="btn btn-success">success</button></div>`;
